src/test/java/dev/openfeature/sdk/ImmutableContextTest.java

```diff
@@ -64,4 +65,27 @@ void mergeShouldReturnAllTheValuesFromTheContextWhenOverridingContextIsNull() {
         assertEquals("targeting_key", merge.getTargetingKey());
         assertArrayEquals(new Object[]{"key1", "key2"}, merge.keySet().toArray());
     }
+
+    @DisplayName("Merge should retain subkeys from the existing context when the overriding context has the same targeting key")
+    @Test
+    void mergeShouldRetainItsSubkeysWhenOverridingContextHasTheSameKey() {
+        HashMap<String, Value> attributes = new HashMap<>();
```

Comment:
Something minor about formatting. Let's have some spaces to separate initializations, method invocations and assertions. Given the length of the test case, current formating makes it little difficult to read :)